core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -6396,6 +6396,36 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
           }
         }
     }
+
+  def render(zio: ZIO.Erased): String =
+    zio match {
+      case Sync(_, _)                    => s"Sync()"
```

Comment:
it's probably best to just match on the type:
```suggestion
      case _: Sync[_, _, _]                    => "Sync()"
```